lib/Onyx.js

```diff
@@ -1025,17 +1028,32 @@ function initializeWithDefaultKeyStates() {
  * Storage.setItem() from Onyx.clear() will have already finished and the merged
  * value will be saved to storage after the default value.
  *
+ * @param {Array} keysToPreserve is a list of ONYXKEYS that should not be cleared with the rest of the data
  * @returns {Promise<void>}
  */
-function clear() {
+function clear(keysToPreserve = []) {
     return getAllKeys()
         .then((keys) => {
-            _.each(keys, (key) => {
-                const resetValue = lodashGet(defaultKeyStates, key, null);
-                cache.set(key, resetValue);
-                notifySubscribersOnNextTick(key, resetValue);
+            const defaultKeys = _.keys(defaultKeyStates);
+
+            // We're clearing out (setting to null) all keys aren't being preserved or have defaults
+            const keysToClear = _.without(keys, ..._.union(keysToPreserve, defaultKeys));
+            const keyValuesToClear = _.map(keysToClear, key => [key, null]);
+
+            // If keysToPreserve contains a key that also has a default key state, we'll preserve it rather than reset
+            // it to the default.
```

Comment:
Can you please update this comment to replace the two instances of "it" with what the reference is? It's a little ambiguous and makes it hard to understand. The comment could also do a little bit better job of explaining _why_ we want this behavior and the intention behind it. Why do we handle the conflict between keysToPreserve and defaultKeyStates this way as opposed to some other way?